### PR TITLE
feat(shorebird_cli): add `--dry-run` and `--force` to `shorebird patch`

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat(shorebird_cli): add --dry-run and --force to patch

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/patch_command.dart [hunk 1]
```diff
@@ -94,6 +106,14 @@ class PatchCommand extends ShorebirdCommand
       return ExitCode.software.code;
     }
 
+    final force = results['force'] == true;
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/patch_command.dart:109]: Looks fine, but isn't this also solved by adding default: false to the addFlag?

Comment (eseidel) [reply packages/shorebird_cli/lib/src/commands/patch_command.dart:109]: Then it's just results['force']?  Donno which is clearer. 🤷

File: packages/shorebird_cli/test/src/commands/patch_command_test.dart [hunk 2]
```diff
@@ -433,9 +452,6 @@ Please create a release using "shorebird release" and try again.
 
     test('throws error when release artifact cannot be retrieved.', () async {
       const error = 'something went wrong';
-      when(
```
Comment (eseidel) [inline packages/shorebird_cli/test/src/commands/patch_command_test.dart:436]: Why are these no longer needed?

**Review by eseidel — APPROVED** [review]

**Review by bryanoltman — APPROVED** [review]

File: packages/shorebird_cli/lib/src/commands/patch_command.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch_command.dart:109]: I tried this – `results['force']` is still null if no arg is provided.

Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/patch_command.dart:109]: `default: false` is the default but accessing arg flags is `dynamic` which is why I wrote it this way.

File: packages/shorebird_cli/test/src/commands/patch_command_test.dart (hunk 2 above)
Comment (felangel) [reply packages/shorebird_cli/test/src/commands/patch_command_test.dart:436]: They are the default stubs so they were unnecessary

**Pull request merged**